src/msca/metrics/main.py

```diff
@@ -0,0 +1,475 @@
+from typing import Optional, List, Union
```

Comment:
I don't think we need these. We can just use `list`, `|` and `| None`. These are all available after 3.10